libs/labelbox/src/labelbox/schema/dataset.py

```diff
@@ -912,3 +913,104 @@ def _convert_items_to_upsert_format(self, _items):
+                >>>    # Get all IAM integrations
+                >>>    iam_integrations = client.get_organization().get_iam_integrations()
+                >>>    
+                >>>    # Get IAM integration id
+                >>>    iam_integration_id = [integration.uid for integration
+                >>>      in iam_integrations
+                >>>      if integration.name == "My S3 integration"][0]
+                >>>
+                >>>   # Set IAM integration for integration id
+                >>>   dataset.set_iam_integration(iam_integration_id)
+                >>>
+                >>>    # Get IAM integration object
+                >>>    iam_integration = [integration.uid for integration
+                >>>      in iam_integrations
+                >>>      if integration.name == "My S3 integration"][0]
+                >>>
+                >>>   # Set IAM integration for IAMIntegrtion object
+                >>>   dataset.set_iam_integration(iam_integration)
+          """
+
+          iam_integration_id = iam_integration.uid if isinstance(iam_integration, IAMIntegration) else iam_integration
+
+          query = """
+            mutation SetSignerForDatasetPyApi($signerId: ID!, $datasetId: ID!) {
+                setSignerForDataset(
+                    data: { signerId: $signerId }
+                    where: { id: $datasetId }
+                ) {
+                    id
+                    signer {
+                        id
+                    }
+                }
+            }
+         """
+
+          response = self.client.execute(query, {"signerId": iam_integration_id, "datasetId": self.uid})
+
+          if not response:
+            raise LabelboxError(f"Can't set IAM integration {iam_integration_id}")
+          
+          try:
+            iam_integration_id = response.get("setSignerForDataset", {}).get("signer", {})["id"]
+            return [integration for integration 
+                    in self.client.get_organization().get_iam_integrations() 
+                    if integration.uid == iam_integration_id][0]
+          except:
+            raise LabelboxError(f"Can't retrieve IAM integration {iam_integration_id}")
+              
+    def remove_iam_integration(self) -> None:
+        """
+        Unsets the IAM integration for the dataset.
+
+        Args:
+            None
+
+        Returns:
+            None
+
+        Raises:
+            LabelboxError: If the IAM integration can't be unset.
+
+        Examples:
+            >>> dataset.remove_iam_integration()
+        """
+
+        query = """
+        mutation DetachSignerPyApi($id: ID!) {
+            clearSignerForDataset(where: { id: $id }) {
+                id
+                signer {
+                    id
+                }
+            }
+        }
+        """
+
+        response = self.client.execute(query, {"id": self.uid})
+
+        if response:
+            return response.get("clearSignerForDataset", {}).get("signer")
```

Comment:
Is the value signer always None? If so, I'd simplify it as:
if not response:
   raise LabelboxError
 <end fo function>
 
Also, if signer is always none, why specify its return value in the gql query?

Reply:
Done.

I've updated the error to 
```raise ResourceNotFoundError(Dataset, {"id": self.uid})```
since response is typically null if the dataset has been deleted.